### PR TITLE
Add additional raft metrics

### PR DESCRIPTION
**Pull request by freddygv**
New metrics were added to raft in [raft-316](https://github.com/hashicorp/raft/pull/316).

This PR re-vendors raft and updates Consul documentation to include the new metrics.

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -188,6 +188,8 @@ github.com/hashicorp/net-rpc-msgpackrpc v0.0.0-20151116020338-a14192a58a69 h1:lc
 github.com/hashicorp/net-rpc-msgpackrpc v0.0.0-20151116020338-a14192a58a69/go.mod h1:/z+jUGRBlwVpUZfjute9jWaF6/HuhjuFQuL1YXzVD1Q=
 github.com/hashicorp/raft v0.0.0-20180817181211-da92cfe76e0c h1:V0ncQQu4St5edf3p566gGAU8jxvrWdjny9dIFl2aV/s=
 github.com/hashicorp/raft v0.0.0-20180817181211-da92cfe76e0c/go.mod h1:DVSAWItjLjTOkVbSpWQ0j0kUADIvDaCtBxIcbNAQLkI=
+github.com/hashicorp/raft v1.0.1-0.20190409200437-d9fe23f7d472 h1:9EPzHJ1bJFaFbGOz3UV3DDFmGYANr+SF+eapmiK5zV4=
+github.com/hashicorp/raft v1.0.1-0.20190409200437-d9fe23f7d472/go.mod h1:DVSAWItjLjTOkVbSpWQ0j0kUADIvDaCtBxIcbNAQLkI=
```
Comment (i0rek) [inline go.sum]: raft is mentioned multiple times here, maybe `go mod tidy` helps?

Comment (mkeeler) [reply go.sum]: I think this is normal. 1 hash is for raft and a second is for the go.mod of raft? If you look at a go.sum all things seem to be duplicated like this although I don't know the underlying reason.

Comment (i0rek) [reply go.sum]: but there is also two more mentions above these new entries.

Comment (mkeeler) [reply go.sum]: Ignore me. Apparently my sight is worse than I thought. There is the v0.0.0 ones above this.

Comment (freddygv) [reply go.sum]: Yep, tidy cleaned up the v0.0.0 entries. Thanks!

**Pull request merged**